tlspuffin/src/protocol.rs

```diff
@@ -603,6 +602,79 @@ impl Extractable<TLSProtocolTypes> for ClientHelloPayload {
     }
 }
 
+impl Extractable<TLSProtocolTypes> for ClientExtension {
```

Comment:
Are those modifications really related to the PR?

Reply:
Yes, in order to query the NamedGroup in the unit test, I added the abilty to extract NamedGroup from the client hello.